### PR TITLE
Multi-DDL statements shouldn't be run in txns

### PR DESCRIPTION
**Pull request by ianjevans**
This PR clarifies that CockroachDB doesn't support fully atomic multi-DDL statements within transactions, and that users should use single-statement transactions when executing schema changes.

It also removes the examples of failing multi-statement transactions.

Fixes DOC-6948.
Fixes DOC-6949.

### CODE REVIEW

**Comment by ianjevans** [comment]
After getting approval on the language changes in v23.1, I will backport the changes to v22.2 and v22.1.

**Comment by ianjevans** [comment]
@dikshant Just a ping on reviewing this. Sounds like some customers are hitting this issue.

**Comment by dikshant** [comment]
LGTM!

**Review by dikshant — APPROVED** [review]

**Review by gemma-shay — APPROVED** [review]
lgtm!

**Pull request merged**
